content/automate/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/pre_retirement.rb

```diff
@@ -1,10 +1,35 @@
 #
 # Description: This method stops a cloud Instance
 #
+module ManageIQ
+  module Automate
+    module Cloud
+      module VM
+        module Retirement
+          module StateMachines
+            module Methods
+              class PreRetirement
+                def initialize(handle = $evm)
+                  @handle = handle
+                end
 
-vm = $evm.root['vm']
-if vm && vm.power_state == 'on'
-  ems = vm.ext_management_system
-  $evm.log('info', "Stopping Instance <#{vm.name}> in EMS <#{ems.try(:name)}>")
-  vm.stop if ems
+                def main
+                  vm = @handle.root['vm']
+                  if vm && vm.power_state == 'on'
+                    ems = vm.ext_management_system
+                    @handle.log('info', "Stopping Instance <#{vm.name}> in EMS <#{ems.try(:name)}>")
```

Comment:
@pkomanek Can you issue the log message only if there is an ems? You can remove the try from the log message in that case.
something like 
if ems
  @handle.log('info', "Stopping Instance <#{vm.name}> in EMS <#{ems.name}>")
  vm.stop
end